netkan/netkan/metadata.py

```diff
@@ -63,3 +63,39 @@ def sqs_message(self):
             'MessageGroupId': '1',
             'MessageDeduplicationId': md5(self.contents.encode()).hexdigest()
         }
+
+
+class Ckan:
+
+    def __init__(self, filename=None, contents=None):
+        if filename:
+            self.filename = Path(filename)
+            self.contents = self.filename.read_text()
+        else:
+            self.contents = contents
+        self._raw = json.loads(self.contents)
+
+    def __getattr__(self, name):
+        if name in self._raw:
+            return self._raw[name]
+        if name == 'kind':
+            return self._raw.get('kind', 'package')
+        raise AttributeError
+
+    # These are properties, but they don't have @property because Python says:
+    #   TypeError: argument of type 'property' is not iterable
+    # ... which doesn't make sense because a property can still return an iterable type.
+
```

Comment:
@property is a decorator, it allows a class function to appear as a property. There might be something in that, but if this works we can refactor later if it becomes annoying.

Reply:
On second thought, that comment was just venting and doesn't belong in the code long term. I'll remove it.